### DISCUSSION TITLE
Question: How can I configure multiple rocks configurations?

### DISCUSSION THREAD

**Discussion by ry-grand** [lumen-oss/rocks.nvim#602] [Q&A] [answered]
I am relatively new to rocks.nvim and am in the process of transitioning my previous configuration(s) to use rocks. However, I am uncertain about how rocks handles multiple neovim configurations, or if it can at all. It would be beneficial if this were possible. For instance, my default configuration in .config/nvim with init.lua and rocks.toml works perfectly. However, when I switch to .config/nvim-dev, running 'Rocks sync' requires the removal of all plugins from the default rocks.toml file before installing the plugins specified in the new rocks.toml file. Although it is simple to revert, by then using 'Rocks sync' when using the default config, switching between them is inefficient and removing and reinstalling plugins from each respective configuration takes much too long. if I were to be able to separate the installs of the plugins based on the configuration entirely, even if it required duplicates, that would be okay, as I simply want both to have the plugins at their disposal. If there are any suggestions or oversights I'm making, I would really appreciate some help. Thanks ❤️ .

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

rocks.nvim uses the standard directories configured by the `:h NVIM_APPNAME` environment variable.
So if you set `NVIM_APPNAME` for your nvim-dev config, rocks.nvim will install plugins into a separate tree than the one for your main config.

**Reply by ry-grand** [reply]
Thanks for the quick response. At first I was confused because I had done this, however it turns out that in my init.lua file, where i specified my rocks_path, I had set it to my other configurations rocks file.

had nvim-dev set to:
```lua
local rocks_config = { rocks_path = "/home/grand/.local/share/nvim/rocks", luarocks_binary = "/home/grand/.local/share/nvim/rocks/bin/luarocks" }
vim.g.rocks_nvim = rocks_config
package.path = package.path .. ";" .. rocks_config.rocks_path.."/share/lua/5.1/?.lua"..";".. rocks_config.rocks_path.."/share/lua/5.1/?/init.lua"
package.cpath = package.cpath .. ";" .. rocks_config.rocks_path.."/lib/lua/5.1/?.so"..";".. rocks_config.rocks_path.."/lib64/lua/5.1/?.so"
```
So now, I actually followed the instructions for the init.lua on the home page😆
This was because I just wanted to simplify my original configuration, however since I copied it over of course it wasn't going to work
Thanks for the help and am really enjoying rocks. All works perfectly now.
